docs/actions.md

```diff
@@ -1,182 +1,129 @@
 # Actions
 
-Use [actions](/docs/api.md#actions) to update the state.
+Actions are functions which take the current [state](/docs/state.md) and return a partial state or a [thunk](#thunks). Actions represent the intent to manipulate the state tree.
 
-```jsx
-app({
-  state: "Hi.",
-  view: (state, actions) =>
-    <h1 onclick={actions.ucase}>
-      {state}
-    </h1>,
-  actions: {
-    ucase: state => state.toUpperCase()
-  }
-})
-```
-
-An action must return a partial state or a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that resolves to a partial state. See [Side Effects](#side-effects).
+[Try it Online](https://codepen.io/hyperapp/pen/qRMEGX?editors=0010)
 
 ```jsx
 app({
   state: {
-    count: 0,
-    maxCount: 10
+    text: "Hello!",
+    defaultText: "Nice weather today."
   },
-  view: (state, actions) =>
-    <main>
+  view: (state, { setText }) =>
+    <div>
       <h1>
-        {state.count}
+        {state.text.trim() === ""
+          ? state.defaultText
+          : state.text}
       </h1>
-      <button onclick={actions.up}>+</button>
-    </main>,
+      <input
+        autofocus
+        value={state.text}
+        oninput={e => setText(e.target.value)}
+      />
+    </div>,
   actions: {
-    up: ({ count, maxCount }) => ({
-      count: count + (maxCount > count ? 1 : -maxCount)
-    })
+    setText: (state, actions, text) => ({ text })
```

Comment:
Just a tiny nitpick how this is written.  It feels inconsistent with everything below being written as:

```js
setText(state, actions, text) {
  return { text }
}
```

Does/can prettier have a preference? Or should they just coexist?